runners/spark/src/main/java/org/apache/beam/runners/spark/SparkPipelineRunner.java

```diff
@@ -212,6 +246,22 @@ public PortablePipelineResult run(RunnerApi.Pipeline pipeline, JobInfo jobInfo)
             pipelineOptions.as(MetricsOptions.class),
             result);
     metricsPusher.start();
+    if (pipelineOptions.getEventLogEnabled()) {
+      eventLoggingListener.onApplicationStart(
+          new SparkListenerApplicationStart(
+              jobInfo.jobId(),
```

Comment:
There are a number of (job or app) * (name or id) fields. It will be misleading to users if one of these fields is used in place of another. I think the following mapping is more accurate:

```java
appName = pipelineOptions.as(ApplicationNameOptions.class).getAppName();
appId = jsc.getConf().getAppId();
appAttemptId = "1";
```

Reply:
I don't get this part. What do you mean?

Reply:
I am referring to certain fields in the Spark API that you have filled in incorrectly. Here (in comments) are the values I think you should use instead:

```java
EventLoggingListener(
  String appId /* jsc.getConf().getAppId() */, 
  java.net.URI logBaseDir, 
  SparkConf sparkConf, 
  org.apache.hadoop.conf.Configuration hadoopConf)

SparkListenerApplicationStart(
  String appName /* pipelineOptions.as(ApplicationNameOptions.class).getAppName() */, 
  scala.Option<String> appId /* jsc.getConf().getAppId() */, 
  long time, 
  String sparkUser, 
  scala.Option<String> appAttemptId /* "1" */, 
  scala.Option<scala.collection.Map<String,String>> driverLogs)
```

Reply:
Thank you, it's done.